redis-http-connector/main.go

```
@@ -0,0 +1,130 @@
+			if err != nil {
+				conn.errorHandler(ctx, err)
+			} else {
+				defer response.Body.Close()
+				body, err := ioutil.ReadAll(response.Body)
+				if err != nil {
+					conn.errorHandler(ctx, err)
+				} else {
+					if success := conn.responseHandler(ctx, string(body)); success {
+						conn.logger.Info("Message sending to response successful")
+					}
+				}
+			}
+		}
+	}()
+	conn.logger.Info("Redis consumer up and running!")
+	<-forever
+}
+
+func (conn redisConnector) errorHandler(ctx context.Context, err error) {
+	if len(conn.connectordata.ErrorTopic) > 0 {
+		err = conn.rdbConnection.RPush(ctx, conn.connectordata.ErrorTopic, err.Error()).Err()
+		if err != nil {
+			conn.logger.Error("Failed to add message in error topic",
+				zap.Error(err),
+				zap.String("source", conn.connectordata.SourceName),
+				zap.String("message", err.Error()),
+				zap.String("topic", conn.connectordata.ErrorTopic))
+		}
+	} else {
+		conn.logger.Error("message received to add to error topic list, but no error topic was set",
+			zap.String("message", err.Error()),
+			zap.String("source", conn.connectordata.SourceName),
+			zap.String("http endpoint", conn.connectordata.HTTPEndpoint))
+	}
+}
+
+func (conn redisConnector) responseHandler(ctx context.Context, response string) bool {
+	if len(conn.connectordata.ResponseTopic) > 0 {
+		err := conn.rdbConnection.RPush(ctx, conn.connectordata.ResponseTopic, response).Err()
+		if err != nil {
+			conn.logger.Error("failed to push response to from http request to topic",
+				zap.Error(err),
+				zap.String("topic", conn.connectordata.ResponseTopic),
+				zap.String("source", conn.connectordata.SourceName),
+				zap.String("http endpoint", conn.connectordata.HTTPEndpoint))
+			return false
+		}
+	}
+	return true
+}
+
+func main() {
+
+	logger, err := zap.NewProduction()
+	if err != nil {
+		log.Fatalf("Can't initiate zap logger :%v", err)
+	}
+	defer logger.Sync()
+
+	connectordata, _ := common.ParseConnectorMetadata()
```

We should not ignore this error?